src/main/java/io/supertokens/storageLayer/StorageLayer.java

```diff
@@ -272,9 +272,12 @@ public static void loadAllTenantStorage(Main main, TenantConfig[] tenants)
                         if (((StorageLayer) main.getResourceDistributor()
                                 .getResource(key.getTenantIdentifier(), RESOURCE_KEY)).storage !=
                                 ((StorageLayer) existingStorageMap.get(key)).storage) {
```

Comment:
change the condition that if it's being used by any tenant, keep it, else close it